### PR TITLE
[IMPROVE] Sync Profile implementations

### PR DESCRIPTION
**Pull request by EugeneButusov**
done.

### CODE REVIEW

File: Soomla/Assets/Plugins/Soomla/Profile/SoomlaProfile.cs [hunk 1]
```diff
@@ -415,7 +451,53 @@ public class SoomlaProfile
...
+		/// <param name="caption">The sub-headline for the link which will be integrated in the story.</param>
+		/// <param name="description">The description for the link which will be integrated in the story.</param>
+		/// <param name="link">The link which will be integrated into the user's story.</param>
+		/// <param name="picture">a Link to a picture which will be featured in the link.</param>
+		/// <param name="payload">A string to receive when the function returns.</param>
+		/// <param name="reward">The reward which will be granted to the user upon a successful update.</param>
+		public static void UpdateStoryDialog(Provider provider, string name, string caption, string description, string link, string picture, string payload, Reward reward = null) {
+			SocialProvider targetProvider = GetSocialProvider(provider);
+			string userPayload = (payload == null) ? "" : payload;
+			if (targetProvider == null)
```
Comment (vedi) [inline Soomla/Assets/Plugins/Soomla/Profile/SoomlaProfile.cs]: @EugeneButusov, do not forget about "{..}"

File: Soomla/Assets/Plugins/Soomla/Profile/social-providers/FBSocialProvider.cs [hunk 2]
```diff
@@ -150,7 +150,32 @@ public FBSocialProvider ()
...
+			FB.Feed(
+				link: link,
+				callback: (FBResult result) => {
+				
+				if (result.Error != null) {
+					fail(result.Error);
+				}
+				else {
+					SoomlaUtils.LogDebug(TAG, "FeedCallback[result.Text]:"+result.Text);
+					SoomlaUtils.LogDebug(TAG, "FeedCallback[result.Texture]:"+result.Texture);
```
Comment (vedi) [inline Soomla/Assets/Plugins/Soomla/Profile/social-providers/FBSocialProvider.cs:170]: @EugeneButusov, Not "feed", is not it?

File: Soomla/Assets/Plugins/Soomla/Profile/social-providers/FBSocialProvider.cs [hunk 3]
```diff
@@ -161,18 +186,55 @@ public FBSocialProvider ()
...
-
-//			checkPermission("publish_actions", ()=> {
-				FB.Feed(
-					link: link,
-					linkName: name,
-					linkCaption: caption,
-					linkDescription: message,
-					picture: pictureUrl,
-					callback: (FBResult result) => {
+			checkPermission("publish_actions", ()=> {
```
Comment (vedi) [inline Soomla/Assets/Plugins/Soomla/Profile/social-providers/FBSocialProvider.cs:191]: @EugeneButusov, do you really need it again? (Looks like degradation regarding https://github.com/vedi/unity3d-profile/commit/c649e928cdf33ebcf7e54828e2a59683554bd7e3#diff-f636b4f7159b314a47eb7964daef81ca).

Comment (EugeneButusov) [inline Soomla/Assets/Plugins/Soomla/Profile/social-providers/FBSocialProvider.cs:191]: @vedi, this code is calling share dialog to make posting, so it moved at `UpdateStoryDialog`, and `UpdateStory` code re-writed to posting without any user confirmations, and it requires `publish_actions` permission.
I did it according to UserProfile aligning (iOS and Android has the same methods, which can post from dialog or directly).

File: Soomla/Assets/Plugins/Soomla/Profile/SoomlaProfile.cs (hunk 1 above)
Comment (EugeneButusov) [inline Soomla/Assets/Plugins/Soomla/Profile/SoomlaProfile.cs]: @vedi, ok.

File: Soomla/Assets/Plugins/Soomla/Profile/social-providers/FBSocialProvider.cs (hunk 2 above)
Comment (EugeneButusov) [inline Soomla/Assets/Plugins/Soomla/Profile/social-providers/FBSocialProvider.cs:170]: @vedi, according to Android implementation of Facebook provider, `updateStory` and `updateStatus` has the following end-point invocation:
`SimpleFacebook.getInstance().publish(feed, false, new OnPublishListener() {`
The difference is in number of parameters (status update has only message, and story update has several parameters).

**Pull request merged**